src/utils/explorer-api.js

```diff
@@ -33,6 +45,10 @@ export async function getTransactions(accountId = '') {
         }
     ));
     
-    return tx.map(t => ({
-        ...t}))
+    return Promise.all(tx.map(async t => ({
+        ...t,
+        status: Object.keys((await transactionExtraInfo(t)).status)[0]
```

Comment:
this looks super inefficient in terms of number of queries needed to receive info.

@marcinbodnar any reason we have to do it that way vs have explorer API return status? /cc @frol

Reply:
I started doing it using `Wampy` with `nearcore-tx`, but it doesn't work. As far as I figured out, the explorer is not storing `status` in db - it does the same thing queries rpc for every transaction separately:
https://github.com/nearprotocol/near-explorer/blob/master/frontend/src/libraries/explorer-wamp/transactions.ts#L180

It would be better to store it in db, and received with `SELECT` query that already is executed in the wallet, but in this case, the explorer would have to check for tx update even if it's not loaded.

We can make it a little bit more efficient by checking status for every transaction, not action - in this case, we can check if `hash` is the same and apply the same value as in action before.

Reply:
This for sure shouldn't be combined as is with polling every 10 sec. Especially given that there seems to be no machinery to cancel outstanding requests.  One open wallet instance will be easily generating 1 request per second with current implementation.

If @frol doesn't have a solution to avoid making extra queries, I think we should disable polling for now.
Polling can be enabled either after changes on explorer side or with such optimizations:
- don't poll when window is not active in browser
- don't make requests per action (make unique request per tx hash)
- only poll status on the transactions which can still change status (if it either succeeded or failed – it cannot change status anymore)

Also if we have to ship multi-query solution – it's best not to block loading of list before all the tx statuses arrive. We can start displaying list of transactions long before all statuses arrive. Status query for transaction in flight can easily be blocking for 10 seconds as for now. Also same goes for error handling – we shouldn't fail loading of the transaction list just because loading status of one of transactions has failed.

Reply:
It is unfortunate that nearcore RPC does not expose the transaction status in the `chunks` API: https://github.com/nearprotocol/nearcore/issues/1880, so we also struggle on the Explorer side. I will take a look in the near future, but it might be a good reason not to expose that as it might require fetching all the receipts for the transaction to know the status (I have not investigated it yet). In any case, we are going to resolve this on Explorer side once we sync all the receipts (https://github.com/nearprotocol/near-explorer/issues/218)

Reply:
> that as it might require fetching all the receipts for the transaction to know the status (

@frol won't us constantly polling it from client-side be even worse? Do you have ETA for https://github.com/nearprotocol/near-explorer/issues/218?

Reply:
> won't us constantly polling it from client-side be even worse? 

Well, it is worse, you are right.

nearprotocol/near-explorer#218 is the next big item on the Explorer development queue, but it is currently estimated to take a couple of weeks to land it properly.

Reply:
> or with such optimizations

@vgrichina ok, np

Reply:
@vgrichina We can also consider adding `check status` option - add it instead of status for every action, or near `Activity` header so it will check status for all actions in activity block?

Reply:
@vgrichina Done. There is just one problem, I'm not sure why a request for tx status is not working here - it's working fine locally. Do you know what is the reason?